src/Extension/EventListener/DebugHttp.php

```diff
@@ -0,0 +1,166 @@
+use Behat\Behat\EventDispatcher\Event\BackgroundTested;
+use Behat\Behat\EventDispatcher\Event\GherkinNodeTested;
+
+use Behat\Gherkin\Node\TaggedNodeInterface;
+
+use Behapi\Extension\Tools\Debug;
+use Behapi\Extension\Tools\HttpHistory;
+
+/**
+ * Debug http
+ *
+ * Allows to debug a scenario, or, if the debug is activated, to print
+ * a message if a scenario failed for http requests.
+ *
+ * @author Baptiste Clavié <clavie.b@gmail.com>
+ */
+class DebugHttp implements EventSubscriberInterface
+{
+    /** @var HttpHistory */
+    private $history;
+
+    /** @var Debug */
+    private $debug;
+
+    public function __construct(Debug $debug, HttpHistory $history)
+    {
+        $this->debug = $debug;
+        $this->history = $history;
+    }
+
+    /** {@inheritDoc} */
+    public static function getSubscribedEvents()
+    {
+        return [
+            OutlineTested::AFTER => 'debugAfter',
+            ScenarioTested::AFTER => 'debugAfter',
+            BackgroundTested::AFTER => 'debugAfter',
+        ];
+    }
+
+    public function debugAfter(GherkinNodeTested $event): void
+    {
+        if (!$event instanceof AfterTested) {
+            return;
+        }
+
+        // no http tag... still no chocolates
+        if (!$this->hasTag($event, 'http')) {
+            return;
+        }
+
+        $history = $this->history;
+
+        // debug only if tag is present (all debug) or only on test failures
+        if (!$this->hasTag($event, 'debug')) {
+            $history = $this->history;
+
+            if (false === $this->debug->getStatus()) {
+                return;
+            }
+
+            $result = $event->getTestResult();
+
+            if (TestResult::FAILED !== $result->getResultCode()) {
+                return;
+            }
+
+            if (is_iterable($result)) {
+                $history = [];
+                $values = iterator_to_array($this->history);
+
+                foreach ($result as $key => $testResult) {
+                    if (TestResult::FAILED !== $testResult->getResultCode()) {
+                        continue;
+                    }
+
+                    $history[] = $values[$key];
+                }
+            }
+        }
+
+        $this->debug($history);
+    }
+
+    private function debug(iterable $historyTuples): void
+    {
+        foreach ($historyTuples as $tuple) {
+            $this->doDebug($tuple[0], $tuple[1]);
+        }
+    }
+
+    private function doDebug(?RequestInterface $request, ?ResponseInterface $response): void
+    {
+        if (!$request instanceof RequestInterface) {
+            return;
+        }
+
+        $debug = $this->getDebug($request, $response);
+
+        foreach ($debug as $key => $value) {
+            echo "\033[36m| \033[1m$key : \033[0;36m$value\033[0m\n";
+        }
+
+        if ($response instanceof ResponseInterface) {
+            echo "\n";
+            echo (string) $response->getBody();
+        }
+
+        echo "\n";
```

Comment:
todo : add some sort of separator ? otherwise it goes

```
| Request: ...
| Request Content-Type: ....
| Response status: ...
| Response Content-Type: ...

{
...
}
| Request: ...
| Request Content-Type: ....
| Response status: ...
| Response Content-Type: ...

{
...
}
```

Or maybe another (!) line break... ?